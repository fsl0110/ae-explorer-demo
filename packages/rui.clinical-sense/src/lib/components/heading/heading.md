# Heading

### Example

```jsx
<Heading tag="h1" color="red">
  I'm a h1 Heading
</Heading>
```
