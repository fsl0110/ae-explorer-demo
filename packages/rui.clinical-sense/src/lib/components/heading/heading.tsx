import React, {FC, ReactNode, HTMLAttributes} from "react";
import {Tag, Text} from "..";

export interface Props extends HTMLAttributes<HTMLHeadingElement> {
  /**
   * Tag to define type of the heading.
   * @default h1
   */
  tag?: "h1" | "h2" | "h3" | "h4" | "h5" | "h6";

  /**
   * Color of the heading.
   */
  color?: string;

  /**
   * Children of the heading.
   */
  children: ReactNode | ReactNode[];
}

export const Heading: FC<Props> = ({tag = "h1", color, children, ...rest}) => (
  <Tag type={tag} {...rest}>
    <Text color={color}>{children}</Text>
  </Tag>
);

export default Heading;
