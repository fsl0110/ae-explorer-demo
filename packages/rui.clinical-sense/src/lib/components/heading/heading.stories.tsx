import React from "react";
import {storiesOf} from "@storybook/react";
import {Heading} from "./heading";
import heading from "./heading.md";

storiesOf("Typography|Heading", module).add(
  "Heading default",
  () => <Heading>Default Heading with h1 Tag</Heading>,
  {
    notes: {markdown: heading},
  },
);

storiesOf("Typography|Heading", module).add(
  "Heading h1",
  () => <Heading tag="h2">Heading with h2 Tag</Heading>,
  {
    notes: {markdown: heading},
  },
);
