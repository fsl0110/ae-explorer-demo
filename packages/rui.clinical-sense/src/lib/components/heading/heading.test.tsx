import React from "react";
import {shallow} from "enzyme";
import toJson from "enzyme-to-json";
import {Heading, Props} from "./heading";

const testCases: Array<[string, Props]> = [
  [
    "Heading h1",
    {
      children: "some text",
    },
  ],
  [
    "Heading h1",
    {
      tag: "h1",
      children: "some text",
    },
  ],
  [
    "Heading h2",
    {
      tag: "h2",
      children: "some text",
    },
  ],
  [
    "Heading h3",
    {
      tag: "h3",
      children: "some text",
    },
  ],
];

describe("<Heading>", () => {
  testCases.forEach(([item, options]) => {
    it(item, () => {
      expect(toJson(shallow(<Heading {...options} />))).toMatchSnapshot();
    });
  });
});
