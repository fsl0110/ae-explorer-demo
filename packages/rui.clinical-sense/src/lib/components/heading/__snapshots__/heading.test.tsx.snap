// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading> Heading h1 1`] = `
<Tag
  type="h1"
>
  <Text>
    some text
  </Text>
</Tag>
`;

exports[`<Heading> Heading h1 2`] = `
<Tag
  type="h1"
>
  <Text>
    some text
  </Text>
</Tag>
`;

exports[`<Heading> Heading h2 1`] = `
<Tag
  type="h2"
>
  <Text>
    some text
  </Text>
</Tag>
`;

exports[`<Heading> Heading h3 1`] = `
<Tag
  type="h3"
>
  <Text>
    some text
  </Text>
</Tag>
`;
